Adapter_Engine/Create/BHoMAdapter.cs

```diff
@@ -0,0 +1,163 @@
+                    }
+                }
+                else
+                {
+                    if (parameter.ParameterType.IsValueType)
+                        arguments.Add(Activator.CreateInstance(parameter.ParameterType));
+                    else
+                        arguments.Add(null);
+                }
+            }
+
+
+            return constructor.Invoke(arguments.ToArray()) as IBHoMAdapter;
+        }
+
+        /***************************************************/
+
+        [Description("Creates an adapter of the specified type. Method makes use of the constructor with the largest number of arguments and passes the provided parameters to that method.")]
+        [Input("adapterType", "The type of adapter to create.")]
+        [Input("parameters", "Arguments expected by the adapter constructor with the largest number of arguments.")]
+        [Output("adapter", "The created adapter of the specified type.")]
+        public static IBHoMAdapter BHoMAdapter(Type adapterType, List<object> parameters)
+        {
+            if (adapterType == null)
+            {
+                Reflection.Compute.RecordError("Provided adapterType is null. Can not create adapter.");
+                return null;
+            }
+
+            if (parameters == null)
+            {
+                Reflection.Compute.RecordError("Provided parameters is null. Can not create adapter.");
+                return null;
+            }
+
+            if (!typeof(IBHoMAdapter).IsAssignableFrom(adapterType))
+            {
+                Reflection.Compute.RecordError("The provided type is not an adapter type. Can not create the adapter.");
+                return null;
+            }
+
+            //Get the constructor with the largest number of arguments
+            ConstructorInfo constructor = adapterType.GetConstructors().OrderByDescending(c => c.GetParameters().Length).First();
+            ParameterInfo[] parameterInfo = constructor.GetParameters();
+
+            if (parameterInfo.Length != parameters.Count)
+            {
+                Reflection.Compute.RecordError($"The provided number of parameters does not match the number of arguments of the constructor with the largest number of inputs. Expecting {parameterInfo.Length} number of arguments and was given {parameters.Count}. Can not create the adapter.");
+                return null;
+            }
+
+            for (int i = 0; i < parameterInfo.Length; i++)
+            {
+                if (parameters[i] == null)
+                {
+                    if (parameterInfo[i].ParameterType.IsValueType)
+                    {
+                        parameters[i] = Activator.CreateInstance(parameterInfo[i].ParameterType);
+                        Reflection.Compute.RecordWarning($"A null argument was provided to an input expecting a value type. A default value of {parameters[i]} has been used in place of the null for the parameter {parameterInfo[i].Name}.");
```

Comment:
What if `null` is a valid input for an adapter constructor? How will this affect its creation?

Reply:
The idea is that this should cover that case.

My logic was "if null and can be null -> fine. If null and can not be null (value type), instantiate a default value of that type"
Basically trying to filter out any non-nullable types and replace the null with a default value of that type, so in case an adapter expected for example an int, then it would replace the null with a 0 etc.

Reply:
The two options I can think of would be to replace the warning and default value creation with an abortion of the method and raising an error, or to just pass in the null to the constructor that probably would crash

Reply:
I'm happy with default value creation for now - just wanted to check what the rationale was, so I'm happy to approve this PR based on that.